### DISCUSSION TITLE
How to read compound dataset with structs which has a variable string length?

### DISCUSSION THREAD

**Discussion by Julian9620** [LiorBanai/HDF5-CSharp#153] [Q&A]
E.g.:
Name       | Type                                                                                                                          | Array Size
Comment | String, length = variable, padding=H5T_STR_NULLTERM, cset = H5T_CSET_ASCII | 1

A example file does not exists so far.

The current implementation uses the Hdf5.ReadCompounds<my_struct>() methode. Is it better to read a list<object> instead of a list<struct>?

**Comment by LiorBanai** [comment]
Many requested read variable data. I'll see what I can do.
do you have example file for me to test?

**Reply by Julian9620** [reply]
No, unfortunately not. In my case it is for the future.

I see also a mistake in my question: It should be: "Is it better to read a list<object> instead of a list<struct>? Because i saw there is also the possibility to create an object with string properties without the byte length.
